src/components/GasOptions/GasOptions.component.js

```diff
@@ -0,0 +1,200 @@
+      const fee = new BigNumber(gasLimit).times(gweiToWei(new BigNumber(gasPriceGwei)));
+      this.props.onChange(fee, new BigNumber(gasLimit), new BigNumber(gasPriceGwei));
+    }
+  }
+
+  onFocusNumberInput(input) {
+    if (this.state[input] === '0') {
+      this.setState({ [input]: '' });
+    }
+  }
+
+  onBlurNumberInput(input) {
+    if (this.state[input] === '') {
+      this.setState({ [input]: '0' });
+    }
+  }
+
+  handleGasPriceChange(e) {
+    const { value } = e.target;
+    // allow an empty input to represent 0
+    if (value === '') {
+      this.setState({ gasPriceGwei: new BigNumber('0'), gasPriceGweiInput: '' });
+    }
+
+    // don't update if invalid regex
+    // (numbers followed by at most 1 . followed by at most 9 decimals)
+    if (!gweiRegex.test(value)) return;
+
+    // don't update if a single gas is an infeasible amount of Ether
+    // (> 100x entire circulating supply as of Aug 2018)
+    if (!isNaN(value) && Number(value) > 10000000000000000000) return;
+
+    // update the input (this could be an invalid number, such as '12.')
+    this.setState({ gasPriceGweiInput: value });
+
+    // update actual gwei if it's a real number
+    if (!isNaN(value)) {
+      this.setState({ gasPriceGwei: new BigNumber(value) });
+    }
+  }
+
+  handleGasLimitChange(e) {
+    const { value } = e.target;
+    // allow an empty input to represent 0
+    if (value === '') {
+      this.setState({ gasLimitInput: '', gasLimit: 0 });
+    }
+
+    // only allow whole numbers
+    if (!gasLimitRegex.test(value)) return;
+
+    // don't allow infeasible amount of gas
+    // (gas limit per block almost never exeeds 10 million as of Aug 2018  )
+    const ONE_HUNDRED_MILLION = 100000000;
+    if (value > ONE_HUNDRED_MILLION) return;
+
+    this.setState({ gasLimitInput: value, gasLimit: parseInt(value, 10) });
+  }
+
+  handleOptionChange(type) {
+    const { gasStatistics } = this.props;
+    if (type === 'manual' || !gasStatistics) {
+      const { defaultGasLimit, defaultGasPrice } = this.props;
+      this.setState({
+        gasPriceGweiInput: defaultGasPrice.toString(),
+        gasPriceGwei: defaultGasPrice,
+        gasLimitInput: defaultGasLimit.toString(),
+        gasLimit: defaultGasLimit,
+        option: 'manual',
+      });
+    } else {
+      const suggestedGasPrice = gasStatistics[type] / 10;
+      this.setState({
+        gasPriceGweiInput: suggestedGasPrice.toString(),
+        gasPriceGwei: suggestedGasPrice,
+        option: type,
+      });
+    }
+  }
+
+  render() {
+    const { gasStatistics, defaultGasLimit, defaultGasPrice, intl } = this.props;
+    const { formatMessage } = intl;
+    const { gasLimitInput, gasPriceGweiInput, option } = this.state;
+    let gasOptions = [{ type: 'manual', name: 'Manual' }];
+    if (gasStatistics) {
+      gasOptions = [
+        { type: 'fast', name: `${formatMessage({ id: 'fast' })} ~ ${Math.ceil(gasStatistics.fastWait)}min` },
```

Comment:
Sweet, I think that looks much better
One thing though, for the English intl at least I think it looks better without a space between the `~` and the estimate, so `~3min` instead of `~ 3min`. Let me know what you think